### PR TITLE
rename Schema to Field

### LINKED ISSUE #577: Rename Schema to Field?
Currently we have the class `Schema` for when you need to define more stuff on a field than just the type and default.

But this is a confusing name since not all it's arguments refer to the schema. It'll get more confusing if we want to add more arguments to `Schema/Field`, eg. a more complex alias system #477.

Therefore we should rename it to `Field`.

This would require:
* renaming the current `Field` class, perhaps renaming the entire `fields.py` module.
* `Schema` needs to carry on working for a while but with a deprecation warning

What does everyone thing?

@tiangolo `Schema` was your work originally, are you ok with this?

### PR DESCRIPTION
**Pull request by samuelcolvin**
## Change Summary

Rename `Schema` to `Field`, make it a function to placate mypy.

## Related issue number

fix #577

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.rst` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: pydantic/fields.py [hunk 1]
```diff
@@ -39,14 +40,132 @@
...
+    min_length: int = None,
+    max_length: int = None,
+    regex: str = None,
+    **extra: Any,
+) -> Any:
+    """
+    Used to provide extra information about a field, either for the model schema or complex valiation. Some arguments
+    apply only to number fields (``int``, ``float``, ``Decimal``) and some apply only to ``str``.
+
+    :param default: since the Schema is replacing the field’s default, its first argument is used
```
Comment (tiangolo) [inline pydantic/fields.py]:
```suggestion
    :param default: since this is replacing the field’s default, its first argument is used
```

**Comment by koxudaxi** [comment]
I have checked the code.  It's very good. Thank you.
Also, The deprecated message is very useful for us.

I'm going to update the PyCharm's plugin to support `Field` by version 1 :)

File: pydantic/fields.py [hunk 2]
```diff
@@ -39,14 +40,132 @@
...
+        self.max_length = kwargs.pop('max_length', None)
+        self.regex = kwargs.pop('regex', None)
+        self.extra = kwargs
+
+    def __repr__(self) -> str:
+        attrs = ((s, getattr(self, s)) for s in self.__slots__)
+        return 'FieldInfo({})'.format(', '.join(f'{a}: {v!r}' for a, v in attrs if v is not None))
+
+
+def Field(
```
Comment (dmontagu) [inline pydantic/fields.py]: Just want to put this on the radar before we commit to this approach:
https://github.com/samuelcolvin/pydantic/issues/837

**Comment by samuelcolvin** [comment]
as per #837 `x: Annotated[int, Field(...)/Schema()]` looks great, but I think we all agree that `x: int = Field(...)` should work too.

**Question:** should we change this so that:
* `Field` is the class, not a function 
* `x: Annotated[int, Field(...)]` works

? 

Making `Field` the actual class would be more pythonic and concise (less repeated code [here](https://github.com/samuelcolvin/pydantic/blob/2724c0abe3354918e4d071425bdb7c023d658aab/pydantic/fields.py#L74-L161)). It would also mean that `x: int = Field(...)` would upset mypy, but in that case you'd use `x: Annotated[int, Field(...)]` anyway?

I guess I'm keen to make this change, but open to import from @tiangolo and @dmontagu.

**Comment by tiangolo** [comment]
Hmm, if possible, I would still prefer the `Field` function or an equivalent.

I have lots of `x: int = Schema(...)  # type: ignore` and I've seen others do as well.

And having only `x: Annotated[int, Field(...)]` as an option increases the verbosity enough that I think I would end up resorting to `x: int = Field(...)  # type: ignore`.

**Comment by samuelcolvin** [comment]
Okay, let's leave it for now. I see now reason why we can't start supporting `x: Annotated[int, Field(...)]` in the future with no backwards compatibility problems.

If anyone sees potential backwards compatibility problems, please let me know asap.

**Comment by dmontagu** [comment]
I have no problems with this. I wouldn’t mind waiting to implement the `Annotated` thing to see if it takes off anywhere else.

**Pull request merged**